datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -2659,16 +2660,24 @@ async fn roundtrip_custom_listing_tables_schema() -> Result<()> {
         .infer_schema(&ctx.state())
         .await?;
 
-    ctx.register_table("hive_style", Arc::new(ListingTable::try_new(config)?))?;
+    let listing_table: Arc<dyn TableProvider> = Arc::new(ListingTable::try_new(config)?);
 
-    let plan = ctx
```

Comment:
Can you please leave the existing test case so it is clear there is no loss of coverage, and add the new plan as a new test case?

Reply:
The original test passes for me (where it should fail, after `cargo clean`) on `main` refs:
- `daeb6597a0c7344735460bb2dce13879fd89d7bd`
- `182d5dc5e456322664da921f446018a0549e60bc`

At any rate, the old plan was 'supposed' to be the new plan! I am happy to make the change if I got this wrong, just wanted to double check 😄

Reply:
What I think it best is if the changes to the tests illustrate the changes in the code on the behavior of the code

If you change the test **and** the code in the same PR it is harder to evaluate the impact of your change on behavior. 

For example, if the original query now fails, perhaps you can change it to 
```rust
let err = thing_that_errors.unwrap_err().to_string();
assert_contains!(err, <expected message>)
```

To show that a test that (incorrectly) used to work no longer does